hail/python/hailtop/utils/utils.py

```diff
@@ -158,8 +174,54 @@ async def wait(self):
         self._waiting = True
         if self._n_complete == self._n_submitted:
             self._done.set()
+        if not self._ignore_errors and self._errors:
+            self._done.set()
```

Comment:
The issue is if we remove the self._waiting, then we add the first function to the queue and that runs successfully before we get a chance to add the second function to the queue. At that point, the n_complete will equal the n_submitted and we'll think we're done and done will be set. Then say the second function is long running and we call wait. It will instantly return because the first function set done.